### DISCUSSION TITLE
Virtual Chassis and LAG interfaces

### DISCUSSION THREAD

**Discussion by kotso** [netbox-community/netbox#6409] [Q&A] [answered]
Hello,

I have question how should I correctly manage aggregate devices and lag interfaces.

I have virtual chassis that consists of multiple Juniper Switches. 

When I go to master device in virtual chassis I see all interfaces from virtual chassis that makes sense, so I added lag interfaces to master device (as I did not find option to create an interface of any kind on virtual chassis itself) 

But when I try to configure any non master member interface, I do not see lag interface in list. (lag interface appears only if I create it on member that owns the port Im trying to configure) but issue is that I have lag interfaces that consists of different member switch ports. 

And I did not figure out how to manage that.

**Comment by kotso** [comment, marked as answer]
to make it more clear.

Virtual Chassis: Juniper-VC (Master is member1)

Member devices: 
- Member1 (interface names ge-0/0/X)
- Member2 (interface names ge-1/0/X)

LAG interface created on member1: ae0

Editing ge-0/0/7 - lets me to choose ae0
Editing ge-1/0/7 - don't see ae0

**Reply by candlerb** [reply]
When you open the master (member1), you will see all the interfaces.  Edit the interface ge-1/0/7 here, and then you can make it a member of the LAG.

**Reply by kotso** [reply]
@candlerb thanks, it worked but its a bit tricky.

When editing single interface on dcim/interfaces/XXX/edit/ link, its not working. 

Worked only when I selected 2 interfaces in listing end used edit button to adjust both.

Anyway, thanks. It worked 🙏

**Reply by candlerb** [reply]
Hmm, that sounds wrong.  If you could show a repeatable way to demonstrate this (starting with new devices from scratch) then you can raise it as an issue.  I tested it myself [recently](https://github.com/netbox-community/netbox/discussions/6116#discussioncomment-687134) with v2.11.2 and it worked for me as expected, using the normal interface edit rather than bulk.

**Reply by kotso** [reply]
@candlerb will be adding new set of devices next week and will test from scratch.

Only thing I remember that I did with devices - interfaces has been renamed as device type has interface names "ge-0/0/[0-47] and when in virtual chassis, interface names are renamed to ge-X/0/[0-47] where x is member id.


Anyway, thanks and will test and be back next week.

**Reply by kotso** [reply]
@candlerb I tried on brand new added switches but it did not work.

If it will help you I can give you my dataset and full installation so you can perform some checks. Or if you say what to check, I could do it.

Version Im running is baddbd667961 (v2.11.3)

Running in docker.

**Reply by candlerb** [reply]
If you can give the step-by-step sequence to reproduce the issue, then you can raise it as a github issue.  In particular, can you identify what steps you did differently to the ones I listed [here](https://github.com/netbox-community/netbox/discussions/6116#discussioncomment-687134).

**Reply by kotso** [reply]
@candlerb I do exactly by instruction you sending.

When I press edit on interface of non master switch in masters port listing I do not see lag interfaces.

I see them only if I select 2 or more interfaces under master interface list and only on bulk edit of multiple interfaces I see lag devices.

**Reply by candlerb** [reply]
You're right: I went through the steps I linked to again, and now when I try to edit an interface I cannot choose a LAG parent.

This might have changed in v2.11.3, since the release notes say:

#6312 - Interface device filter should return all virtual chassis interfaces only if device is master

I have raised this as issue #6426
